drawBot/context/baseContext.py

```diff
@@ -1959,13 +1962,22 @@ def appendGlyph(self, *glyphNames):
         _openTypeFeatures = dict(self._openTypeFeatures)
         self._openTypeFeatures = dict(calt=False)
         for glyphName in glyphNames:
-            glyph = font.glyphWithName_(glyphName)
+            if isinstance(glyphName, int):
+                # support glyph indexes
+                glyph = glyphName
+            else:
+                glyph = font.glyphWithName_(glyphName)
             if glyph:
                 self.append(baseString)
                 glyphInfo = AppKit.NSGlyphInfo.glyphInfoWithGlyph_forFont_baseString_(glyph, font, baseString)
                 self._attributedString.addAttribute_value_range_(AppKit.NSGlyphInfoAttributeName, glyphInfo, (len(self) - 1, 1))
             else:
-                warnings.warn("font '%s' has no glyph with the name '%s'" % (font.fontName(), glyphName))
+                if isinstance(glyphName, int):
```

Comment:
Not sure about this change: this basically checks whether the user passed 0 as a GID. Perhaps there should be a check after `glyphInfo = AppKit.NSGlyphInfo.glyphInfoWithGlyph_forFont_baseString_(glyph, font, baseString)`

Reply:
`.notdef` or `0` are both not possible. `.notdef` will result in glyph being `0` (it has to be the first glyph by index)

CoreText will fallback to an other font or the system font and will draw the `.notdef` of the systemfont which is a ? inside a diamond shape. And not the .notdef glyph of the current font...

Reply:
this produces also a different warning message based on the type of the `glypyhName`

Reply:
It is now possible to pass an out-of-range GID and cause an error. `glyphInfo` will be `None` in that case, and `self._attributedString.addAttribute_value_range_(...)` will throw something like this:

`ValueError: NSInvalidArgumentException - NSConcreteMutableAttributedString addAttribute:value:range:: nil value`

Reply:
And also, if you pass a GID of 0, yopu will get this warning:
```
message = "font '{fontName}' has no glyph with index '{glyphName}'"
```
Which makes no sense as the font _does_ contain a glyph with GID == 0.